src/Value.js

```diff
@@ -17,7 +17,7 @@ var Option = React.createClass({
 	render: function() {
 		return (
 			<div className="Select-item">
-				<span className="Select-item-icon" onMouseDown={this.blockEvent} onClick={this.props.onRemove}>&times;</span>
+				<span className="Select-item-icon" onMouseDown={this.blockEvent} onClick={this.props.onRemove} onTouchEnd={this.props.onRemove}>&times;</span>
```

Comment:
No, I tested, the `onClick` does not fire on a touch device.

Add the `onTouchEnd` will make it work.

I also tried to change the `onTouchEnd` handler to 
```
onTouchHanlder: function(e) {
  this.blockEvent(e)
  this.props.onRemove()
}
```
to prevent the input filed from focusing, but seems not work.